cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -146,14 +148,19 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
       LiveView   -> do
 #ifdef UNIX
         let c = llConfiguration loggingLayer
+        -- TODO: This desperately needs to be refactored.
         -- check required tracers are turned on
         let reqtrs = [ ("TraceBlockFetchDecisions",traceBlockFetchDecisions)
                      , ("TraceChainDb",traceChainDB)
                      , ("TraceForge",traceForge)
                      , ("TraceMempool",traceMempool)
                      ]
-            trsinactive = filter (\(_,f) -> not $ traceEnabled (ncTraceConfig nc) f) reqtrs
-        unless (List.null trsinactive) $ do
+
+        trsinactive <- case ncTraceConfig nc of
+                         TracingOff -> return []
+                         TracingOn traceConf -> return $ map (\(tName,f) -> (tName, f traceConf)) reqtrs
```

Comment:
previously, the list only included those tracers which are turned off.

Reply:
My most recent change should now function as before. I'd like to also circle back and propose some further changes here.